app/models/account_secret.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: account_secrets
+#
+#  id          :bigint(8)        not null, primary key
+#  private_key :text
+#  account_id  :bigint(8)        not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+class AccountSecret < ApplicationRecord
```

Comment:
Is this idea here that this class will only ever hold the private key? Might it hold the public key as well? (not as strong an encryption need) Might it hold other encryption-needing values connected to accounts?

It might be nice to do a refactor either with this change or ahead of it that extracts some of the "key handling" (generating pairs, and knowledge of what conditions to do so) either to this class (if public key also moved), or to a model concern or some other new class that could own "key stuff". This is a half baked idea though,  need to look more.

Reply:
> Might it hold the public key as well?

I was asking myself this question. Since a public key is not a secret I decided to leave it in `Account`, but I also understand the reasoning to keep public and private keys close to each other.

> Might it hold other encryption-needing values connected to accounts?

Yes.

Reply:
Makes sense ... I guess if the design goal is "lets have one table where things which need to be encrypted on the account are stored", then this makes sense as-is, its fine that public/private are in different spots, and future hypothetical other-columns could be added.

I wonder if the new table was instead account_keys (or whatever) and held both the public and private, and also the logic around creating/parsing/etc them - that might gain us a nice readability/encapsulation benefit, while also moving the encryption-needing attribute to a different table (as desired in linked issue). Could also open the door for having multiple keys per account, though I don't want to scope creep this too much for now.